forge-ai/src/main/java/forge/ai/ability/AttachAi.java

```diff
@@ -1257,6 +1256,19 @@ public boolean apply(final Card c) {
             });
         }
 
+        // Should not attach things to crewed vehicles that will stop being creatures soon
+        // Equipping in Main 1 on creatures that actually attack is probably fine though
+        // TODO Somehow test for definitive advantage (e.g. opponent low on health, AI is attacking)
+        // to be able to deal the final blow with an enchanted vehicle like that
+        if (attachSource.isAura() || attachSource.isEquipment()) {
+            prefList = CardLists.filter(prefList, new Predicate<Card>() {
+                @Override
+                public boolean apply(Card card) {
+                    return card.getTimesCrewedThisTurn() == 0 || (attachSource.isEquipment() && attachSource.getGame().getPhaseHandler().is(PhaseType.MAIN1, ai));
```

Comment:
Hmn, I think you'll want to check that the aura would really fall off

- _Akki War Paint_ - "Enchant artifact or creature"
- _Rune of Might_ - "Enchant permanent"

Reply:
Yeah, I also thought about that, but couldn't devise a good enough way of checking this :/ Do you have any ideas? My best attempt at this involved checking for "Creature[.YouCtrl]" in ValidTgts, but that's most likely suboptimal.

Reply:
I added my version of the check, though I'm sure it can somehow be done better ^^;
Ideally, would also need to account for whatever the card will become after it stops being a creature, but that sounds quite tricky to do.

Reply:
Right, I think like this we should be at least 90% correct 👍